src/entry.js

```diff
@@ -61,15 +61,20 @@ class Entry {
    * @param {Entry} entry The entry being verified
    * @return {Promise} A promise that resolves to a boolean value indicating if the signature is valid
    */
-  static async verify (identityProvider, entry) {
-    if (!identityProvider) throw new Error('Identity-provider is required, cannot verify entry')
+  static async verify (identities, entry) {
+    if (!identities) throw new Error('identities is required, cannot verify entry')
     if (!Entry.isEntry(entry)) throw new Error('Invalid Log entry')
     if (!entry.key) throw new Error("Entry doesn't have a key")
     if (!entry.sig) throw new Error("Entry doesn't have a signature")
 
+    // verify identity if entry-version > 0
+    const validIdentity = entry.v > 0 ? await identities.verifyIdentity(entry.identity) : true
+    if (!validIdentity) {
+      throw new Error(`${entry.identity.id} is invalid`)
+    }
     const e = Entry.toEntry(entry, { presigned: true })
     const verifier = entry.v < 1 ? 'v0' : 'v1'
-    return identityProvider.verify(entry.sig, entry.key, Entry.toBuffer(e), verifier)
+    return identities.keystore.verify(entry.sig, entry.key, Entry.toBuffer(e), verifier)
```

Comment:
Any reason we can't keep the old api here and call `identities.verify()` directly? I think the sign/verify API is not symmetrical in that verify is in `identities.keystore` and sign is in `identities`. Can we provide `verify()` function directly through `identities`?